Manager/src/main/java/io/raspberrywallet/manager/bitcoin/WalletCrypter.java

```diff
@@ -0,0 +1,55 @@
+package io.raspberrywallet.manager.bitcoin;
+
+import com.google.protobuf.ByteString;
+import com.stasbar.Logger;
+import org.bitcoinj.crypto.KeyCrypterScrypt;
+import org.bitcoinj.wallet.Protos;
+import org.bitcoinj.wallet.Wallet;
+import org.jetbrains.annotations.NotNull;
+import org.spongycastle.crypto.params.KeyParameter;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class WalletCrypter {
+    // These params were determined empirically on a top-range (as of 2014) MacBook Pro with native scrypt support,
+    // using the scryptenc command line tool from the original scrypt distribution, given a memory limit of 40mb.
+    private static final Protos.ScryptParameters SCRYPT_PARAMETERS = Protos.ScryptParameters.newBuilder()
+            .setP(6)
+            .setR(8)
+            .setN(32768)
+            .setSalt(ByteString.copyFrom(KeyCrypterScrypt.randomSalt()))
+            .build();
+
+
+    void decryptWallet(@NotNull Wallet wallet, @NotNull String password) {
+        if (password.length() == 0 || password.length() < 4) {
+            throw new IllegalArgumentException("Bad password. The password you entered is empty or too short.");
+        }
+
+        final KeyCrypterScrypt scrypt = (KeyCrypterScrypt) wallet.getKeyCrypter();
+        checkNotNull(scrypt);   // We should never arrive at this GUI if the wallet isn't actually encrypted.
+        KeyParameter aesKey = scrypt.deriveKey(password);
+        if (wallet.checkAESKey(aesKey)) {
+            wallet.decrypt(aesKey);
+        } else {
+            Logger.err("User entered incorrect password");
+            Logger.err("Wrong password");
```

Comment:
these two log lines are actually the same

Reply:
Right, fixed